libs/sdk-model/src/accessControl/index.ts

```diff
@@ -44,12 +44,47 @@ export const isUserGroupAccess = (obj: unknown): obj is IUserGroupAccess => {
     return !isEmpty(obj) && (obj as IUserGroupAccess).type === "group";
 };
 
+/**
+ * User access specification with granular permissions.
+ *
+ * @alpha
+ */
+export type IUserAccessWithGranularPermissions = IUserAccess & IAccessGranteeWithGranularPermissions;
+
+/**
+ * User group access specification with granular permissions.
+ *
+ * @alpha
+ */
+export type IUserGroupAccessWithGranularPermissions = IUserGroupAccess &
+    IAccessGranteeWithGranularPermissions;
+
+/**
+ * Tests whether the provided object is an instance of {@link IUserAccessWithGranularPermissions} or {@link IUserGroupAccessWithGranularPermissions}.
+ *
+ * @param obj - object to test
+ * @alpha
+ */
+export const isAccessWithGranularPermissions = (
+    obj: unknown,
+): obj is IUserAccessWithGranularPermissions | IUserGroupAccessWithGranularPermissions => {
+    return (
+        !isEmpty(obj) &&
+        (obj as IUserAccessWithGranularPermissions | IUserGroupAccessWithGranularPermissions).permissions !==
+            undefined
+    );
+};
+
 /**
  * Entity having access to the object.
  *
  * @alpha
  */
-export type AccessGranteeDetail = IUserAccess | IUserGroupAccess;
+export type AccessGranteeDetail =
+    | IUserAccess
+    | IUserGroupAccess
+    | IUserAccessWithGranularPermissions
```

Comment:
I would prefer `Granular` instead of `WithGranularPermissions` in type and guard names